crates/driver/src/boundary/settlement.rs

```diff
@@ -245,6 +248,115 @@ impl Settlement {
             solver: self.solver,
         })
     }
+
+    pub fn with_protocol_fees(&mut self, solution: &competition::Solution) {
```

Comment:
nit: slightly nicer interface if it consumes self and returns `Self` (with the adjustments)